product-cli/cmd/cluster/cluster.go

```diff
@@ -0,0 +1,51 @@
+package cluster
+
+import (
+	"github.com/openshift/hypershift/api/v1beta1"
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/hypershift/cmd/cluster/agent"
+	"github.com/openshift/hypershift/cmd/cluster/aws"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	"github.com/openshift/hypershift/cmd/cluster/kubevirt"
+	"github.com/openshift/hypershift/cmd/log"
+)
+
+func NewCreateCommands() *cobra.Command {
+	opts := &core.CreateOptions{
+		Namespace:                      "clusters",
+		Name:                           "example",
+		ReleaseImage:                   "",
+		PullSecretFile:                 "",
+		ControlPlaneAvailabilityPolicy: "SingleReplica",
+		Render:                         false,
+		InfrastructureJSON:             "",
+		InfraID:                        "",
+		ServiceCIDR:                    "172.31.0.0/16",
+		ClusterCIDR:                    "10.132.0.0/14",
+		Wait:                           false,
+		Timeout:                        0,
+		ExternalDNSDomain:              "",
+		AdditionalTrustBundle:          "",
+		ImageContentSources:            "",
+		NodeSelector:                   nil,
+		Log:                            log.Log,
+		NodeDrainTimeout:               0,
+		NodeUpgradeType:                v1beta1.UpgradeTypeReplace,
+		Arch:                           "amd64",
+	}
+	cmd := &cobra.Command{
+		Use:          "cluster",
+		Short:        "Creates basic functional HostedCluster resources",
+		SilenceUsage: true,
+	}
+
+	cmd.PersistentFlags().StringVar(&opts.Namespace, "namespace", opts.Namespace, "A namespace to contain the generated resources")
+	cmd.PersistentFlags().StringVar(&opts.Name, "name", opts.Name, "A name for the cluster")
+
+	cmd.AddCommand(aws.NewCreateCommand(opts))
```

Comment:
let's start with the very minimum for a first pass and include only one platform e.g agent